src/main/java/org/veupathdb/service/eda/ds/plugin/differentialabundance/DifferentialAbundanceVolcanoplotPlugin.java

```diff
@@ -0,0 +1,91 @@
+import org.veupathdb.service.eda.generated.model.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.veupathdb.service.eda.common.plugin.util.RServeClient.useRConnectionWithRemoteFiles;
+
+public class DifferentialAbundanceVolcanoplotPlugin extends AbstractPlugin<DifferentialAbundanceVolcanoplotPostRequest, DifferentialAbundanceVolcanoplotSpec, DifferentialAbundanceComputeConfig> {
+  
+  @Override
+  public String getDisplayName() {
+    return "Volcano plot";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Display fold change vs. significance for a differential abundance analysis.";
+  }
+
+  @Override
+  public List<String> getProjects() {
+    return List.of(AppsMetadata.MICROBIOME_PROJECT);
+  }
+
+  @Override
+  protected ClassGroup getTypeParameterClasses() {
+    return new ClassGroup(DifferentialAbundanceVolcanoplotPostRequest.class, DifferentialAbundanceVolcanoplotSpec.class, DifferentialAbundanceComputeConfig.class);
+  }
+
+  // // I think we can remove this whole thing ?!? there are no variables for the user to select
+  // @Override
+  // public ConstraintSpec getConstraintSpec() {
+  //   return new ConstraintSpec()
+  //     .dependencyOrder(List.of("yAxisVariable", "xAxisVariable"), List.of("overlayVariable"))
+  //     .pattern()
+  //       .element("overlayVariable")
+  //         .required(false)
+  //         .maxValues(8)
+  //         .description("Variable must be a number, or have 8 or fewer values, and be of the same or a parent entity as the X-axis variable.")
+  //     .pattern()
+  //       .element("overlayVariable")
+  //         .required(false)
+  //         .types(APIVariableType.NUMBER, APIVariableType.INTEGER) 
+  //         .description("Variable must be a number, or have 8 or fewer values, and be of the same or a parent entity as the X-axis variable.")
+  //     .done();
+  // }
+
+  // // and this??
+  // @Override
+  // protected void validateVisualizationSpec(DifferentialAbundanceVolcanoplotSpec pluginSpec) throws ValidationException {
+  //   validateInputs(new DataElementSet()
+  //     .entity(pluginSpec.getOutputEntityId())
+  //     .var("overlayVariable", pluginSpec.getOverlayVariable()));
+  // }
+
+  // Not sure about this... is this grabbing the tabular?
+  @Override
+  protected List<StreamSpec> getRequestedStreams(DifferentialAbundanceVolcanoplotSpec pluginSpec) {
+    return List.of(
+      new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, pluginSpec.getOutputEntityId())
+        .setIncludeComputedVars(true)
+    );
+  }
+
+  @Override
+  protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
+    DifferentialAbundanceVolcanoplotSpec spec = getPluginSpec();
+    DifferentialAbundanceStats stats = getComputeResultStats(DifferentialAbundanceStats.class);
+
+    System.out.println(stats);
```

Comment:
You need to convert the stats object to a json string and write it to the output stream.  This is not too hard, but maybe I'll add a utility if we think a direct stream from compute stats out the door will be a common use case?

Reply:
i think its likely to be a common case. i need to think some more about it to be sure, but i could easily see us doing the same for the correlations app(s)